### ISSUE TITLE
incorrect type checking of loop variables

### ISSUE THREAD

**Issue by charles-cooper** [vyperlang/vyper#3212]
### Version Information

* vyper Version (output of `vyper --version`): 0.3.7

the type of a loop variable can take multiple values in a loop. minimal repro:
```vyper
@external
def main():
    for j in range(3):
        x: uint256 = j
        y: uint16 = j
```

thanks to @trocher for reporting

**Comment by charles-cooper** [comment]
this is a subset of https://github.com/vyperlang/vyper/issues/3374, which is fixed by #3375, closing

**Comment by charles-cooper** [comment]
actually, might not be a subset

**Comment by charles-cooper** [comment]
fixed as of #3375
